packages/app/src/cli/services/dev/extension/websocket/handlers.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  EventType,
+  IncomingDispatchMessage,
+  OutgoingDispatchMessage,
+  OutgoingMessage,
+  SetupWebSocketConnectionOptions,
+} from './models.js'
+import {RawData, WebSocket, WebSocketServer} from 'ws'
+import {http, output} from '@shopify/cli-kit'
+import {Duplex} from 'stream'
+
+export function websocketUpgradeHandler(
+  wss: WebSocketServer,
+  options: SetupWebSocketConnectionOptions,
+): (req: http.IncomingMessage, socket: Duplex, head: Buffer) => void {
+  return (request, socket, head) => {
+    if (request.url !== '/extensions') {
+      return
+    }
+    output.debug(`Upgrading HTTP request to a websocket connection`)
+    wss.handleUpgrade(request, socket, head, connectionDoneHandler(wss, options))
+  }
+}
+
+export function connectionDoneHandler(wss: WebSocketServer, options: SetupWebSocketConnectionOptions) {
+  return (ws: WebSocket) => {
+    output.debug(`Websocket connection successfully established`)
+    const connectedPayload = {
+      event: 'connected',
+      data: options.payloadStore.getConnectedPayload(),
+      version: '3',
+    }
+    output.debug(output.content`Sending connected payload: ${output.token.json(connectedPayload)}`)
+    ws.send(JSON.stringify(connectedPayload))
+    ws.on('message', onMessageHandler(wss, options))
+  }
+}
+
+export function onMessageHandler(wss: WebSocketServer, options: SetupWebSocketConnectionOptions) {
```

Comment:
Name is a bit misleading as its not the handler, its a factory that _gives_ you the handler.

Reply:
Yeah, fair.  

I've reviewed all our handlers.  We now use `getSomeHander()` if the function returns a handler, or `someHandler()` if the function is the handler